Assets/Scripts/Models/Time/WorldTime.cs

```diff
@@ -292,15 +292,41 @@ public int Year
         return Math.Abs(time1.Seconds - time2.Seconds) > float.Epsilon;
     }
 
+    public override bool Equals(object obj)
+    {
+        if (obj.Equals(this))
+        {
+            return true;
+        }
+        else if (obj is WorldTime && ((WorldTime)obj) == this)
```

Comment:
technically you just need two ifs statements followed by `return false`. the elses are redundant since if it had been satisfied it would have already returned.